blockchain/v2/reactor.go

```diff
@@ -0,0 +1,103 @@
+package v2
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+type timeCheck struct {
+}
+
+func schedulerHandle(event Event) (Events, error) {
+	switch event.(type) {
+	case timeCheck:
+		fmt.Println("scheduler handle timeCheck")
+	case Event:
+		fmt.Println("scheduler handle testEvent")
+	}
+	return Events{}, nil
+}
+
+func processorHandle(event Event) (Events, error) {
+	switch event.(type) {
+	case timeCheck:
+		fmt.Println("processor handle timeCheck")
+	case Event:
+		fmt.Println("processor handle event")
+	}
+	return Events{}, nil
+}
+
+type Reactor struct {
+	demuxer       *demuxer
+	scheduler     *Routine
+	processor     *Routine
+	tickerStopped chan struct{}
+}
+
+// nolint:unused
+func (r *Reactor) setLogger(logger log.Logger) {
+	r.scheduler.setLogger(logger)
+	r.processor.setLogger(logger)
+	r.demuxer.setLogger(logger)
+}
+
+func (r *Reactor) Start() {
+	r.scheduler = newRoutine("scheduler", schedulerHandle)
+	r.processor = newRoutine("processor", processorHandle)
+	r.demuxer = newDemuxer(r.scheduler, r.processor)
+	r.tickerStopped = make(chan struct{})
+
+	go r.scheduler.start()
+	go r.processor.start()
+	go r.demuxer.start()
+
+	<-r.scheduler.ready()
```

Comment:
Can't we use WaitGroup or https://godoc.org/golang.org/x/sync/errgroup here?

Reply:
Definitely could use wait groups here but didn't want to include inter routine coordination in the routine. I updated the the code s/t the `rdy` is now closed as soon as the routine in ready, this way `<-ready()` will return immediately if the routine is "ready" as the name implies.